mysql.go

```diff
@@ -71,149 +85,76 @@ func (driver *Driver) Initialize(url string) error {
 	if err := db.Ping(); err != nil {
 		return err
 	}
-	driver.db = db
+	drv.db = db
 
-	if err := driver.ensureVersionTableExists(); err != nil {
-		return err
-	}
-	return nil
+	return drv.ensureVersionTableExists()
 }
 
-func (driver *Driver) Close() error {
-	if err := driver.db.Close(); err != nil {
-		return err
-	}
-	return nil
+// Close db connection
+func (drv *Driver) Close() error {
+	return drv.db.Close()
 }
 
-func (driver *Driver) ensureVersionTableExists() error {
-	_, err := driver.db.Exec("CREATE TABLE IF NOT EXISTS " + tableName + " (version bigint not null primary key);")
-	if err != nil {
-		return err
-	}
-	
-	r := driver.db.QueryRow("SELECT data_type FROM information_schema.columns where table_name = ? and column_name = 'version'", tableName)
-	dataType := ""
-	if err := r.Scan(&dataType); err != nil {
-		return err
-	}
-	if dataType != "int" {
-		return nil
-	}
-	_, err = driver.db.Exec("ALTER TABLE " + tableName + " MODIFY version bigint")
-	return err
+// FilenameExtension of migration file.
+func (drv *Driver) FilenameExtension() string {
+	return "sql"
 }
 
-func (driver *Driver) FilenameExtension() string {
-	return "sql"
+// FileTemplate of migration file.
+func (drv *Driver) FileTemplate() []byte {
+	return []byte(`
+-- Each SQL statement MUST end with semicolon (;) FOLLOWED BY NEWLINE !
+-- Whole migration will be executed inside transaction by default.
+-- Place SQL between "-- TXBEGIN" and "-- TXEND" comments for custom transaction:
+--   - you CAN have multiple separate transactions in single migration
+--   - any SQL not wrapped into TXBEGIN - TXEND will be executed without transaction.
+-- Add "-- NOTX" comment above all SQL to disable default migration. NOTE:
+--   it's redundant when TXBEGIN/TXEND is used.
+`)
 }
 
-func (driver *Driver) Migrate(f file.File, pipe chan interface{}) {
-	defer close(pipe)
-	pipe <- f
+// Execute sql
+func (drv *Driver) Execute(sql string) error {
+	_, err := drv.db.Exec(sql)
+	return err
+}
 
-	// http://go-database-sql.org/modifying.html, Working with Transactions
-	// You should not mingle the use of transaction-related functions such as Begin() and Commit() with SQL statements such as BEGIN and COMMIT in your SQL code.
-	tx, err := driver.db.Begin()
-	if err != nil {
-		pipe <- err
-		return
+// Migrate runs migration.
+// It locks schema_migrations table, so concurrent execution is safe.
+func (drv *Driver) Migrate(f file.File) error {
+	if drv.versionTx == nil {
+		return errors.New("migrate must call Lock before Migrate")
```

Comment:
Why don't you lock here instead of returning an error?
There could be an `unlock` added as a defer too.

Reply:
If we could lock/unlock here, there would be no reason for exposing Lock/Unlock methods in driver interface. `migrate` package is responsible for locking